app/views/helpers/AdditionalInformationRequiredHelper.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.helpers
+
+import config.AppConfig
+import models.DeclarationType.{CLEARANCE, DeclarationType}
+import models.declaration.ProcedureCodesData
+import play.api.mvc.Call
+import play.api.i18n.Messages
+import play.twirl.api.Html
+import uk.gov.hmrc.govukfrontend.views.Aliases.HtmlContent
+import views.html.components.gds.{exportsInsetText, link, paragraphBody}
+
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class AdditionalInformationRequiredHelper @Inject()(appConfig: AppConfig, paragraphBody: paragraphBody, insetText: exportsInsetText, link: link) {
+  def getBodyContent(decType: DeclarationType, mayBeProcedureCode: Option[ProcedureCodesData])(implicit messages: Messages): Html =
+    (decType, mayBeProcedureCode.flatMap(_.procedureCode)) match {
+      case (CLEARANCE, _)                => getBodyContentForClearanceOr1040()
```

Comment:
I did not try myself but just to be safe, considering the order of the cases, are you sure that if you have, for instance, Clearance and 1042 this case will not be considered?

Reply:
It will be caught by this case yes (that is the desired behavior, if on clearance journey then the PC does not matter) :slightly_smiling_face:

Reply:
I know you like pattern matching :smile: but tbh It's quite tricky to get in this form. At the end it's just a simple `if`, isn't it?
```
val is1040 = mayBeProcedureCode.exists(_.procedureCode == 1040)
if (is1040 || decType == CLEARANCE) getBodyContentForClearanceOr1040()
else getBodyContentForNot1040(pc)
```

Reply:
But it's subjective, of course. I have no problem with your syntax.

Reply:
I get where you're coming from, but the pattern matching is also handling the optionality of `mayBeProcedureCode.flatMap(_.procedureCode)` (so covering off a default action if something odd happens like user skipping the PC pages). 
But yeah, its all subjective ;)